packages/mimecast/data_stream/archive_search_logs/elasticsearch/ingest_pipeline/default.yml

```
@@ -0,0 +1,155 @@
+---
+description: Pipeline for processing Mimecast archive_search_logs.
+processors:
+  # Generic event/ecs fields we always want to populate.
+  - set:
+      field: ecs.version
+      value: '8.9.0'
+  - rename:
+      field: message
+      target_field: event.original
+  - json:
+      description: Parse 'message' JSON contents into a 'mimecast' object.
+      field: event.original
+      target_field: mimecast
+  - drop:
+      if: ctx.mimecast?.createTime == null || (ctx.mimecast?.data != null && ctx.mimecast.data.isEmpty())
```

```suggestion
      if: ctx.mimecast?.createTime == null || (ctx.mimecast.data != null && ctx.mimecast.data.isEmpty())
```
null check on `mimecast` already done

---

If `mimecast` is null , then the first condition `ctx.mimecast?.createTime == null` returns `false` and because there is `||` the second condition is evaluated and probably `mimecast` null check is still needed?

---

I didn't see the `||` 👍🏼